controllers/articles.js

```diff
@@ -239,6 +267,44 @@ const articles = {
       }
     }
   },
+
+  rateArticle: async (req, res) => {
+    const { slug } = req.params;
+    const { rating } = req.body;
+    try {
+      const result = await article.findOne({ where: { slug } });
+      if (!result) {
+        return res.status(404).send({
+          status: res.statusCode,
+          errorMessage: 'Article not found',
+        });
+      }
+      const ratings = result.ratings == null ? [Number(rating)]
+        : result.ratings.concat(Number(rating));
+
+      const updated = await article.update({ ratings }, {
+        where: { slug },
+        returning: true,
+      });
+
+      const data = updated[1][0].get();
+      data.ratings = getAverageRating(data);
+
+      delete data.id;
+      delete data.deleted;
+      return res.status(200).send({
+        status: res.statusCode,
+        data,
+      });
+    } catch (error) {
+      if (error.message) {
+        return res.status(500).send({
+          status: res.statusCode,
+          errorMessage: error.message,
```

Comment:
Sometimes this error is not descriptive you can look on how to return a good error message.

Reply:
Solved this.